lib/logstash/inputs/jdbc.rb

```diff
@@ -230,11 +232,29 @@ module LogStash module Inputs class Jdbc < LogStash::Inputs::Base
   config :target, :validate => :field_reference, :required => false
 
   attr_reader :database # for test mocking/stubbing
+  attr_reader :last_run_metadata_file_path # path to the file used as last run storage
 
   public
 
   def register
     @logger = self.logger
+
+    if record_last_run
+      if last_run_metadata_path.nil?
```

Comment:
In order to avoid this being a breaking transition, we need to make a one-time copy of the file in `#{ENV['HOME']}/.logstash_jdbc_last_run` to our new `path.data`-based default path. This will ensure that a user with an existing implicit configuration has continuity of behaviour before and after the upgrade.

Reply:
It could be that after an upgrade with `deb`or `rpm` distribution the owner of the old folder change, so the old file can't be deleted.
The idea to implement here is:
-  if the file `#{ENV['HOME']}/.logstash_jdbc_last_run` exists and new one in `path.data` doesn't, then copy and try to delete the old. If the deletion fail, log a WARN informing the user he has to manually delete the old file.
- if exists both the old and new, print a WARN to inform the user and consider the new one to valid for use.

Reply:
@yaauie I swtiched a little bit the flow described above:
if the `#{ENV['HOME']}/.logstash_jdbc_last_run` exists then move in the `path.data` sub-folder. If it can't because of access rights to the old file, then fail with a ConfigurationError so that the user has to intervene, fixing the access right to the old file or maybe moving it manually.